### ISSUE TITLE
Cyclus compilation failing against Cython 0.27

### ISSUE THREAD

**Issue by bam241** [cyclus/cyclus#1420]
```/Users/mouginot/work/app/cyclus/cyclus/build/cyclus/py_agents_.cxx:6166:23: error: out-of-line definition of '__pyx_f___init__CyclusAgentShim' does not match any
      declaration in 'CyclusAgentShim'
void CyclusAgentShim::__pyx_f___init__CyclusAgentShim(CYTHON_UNUSED cyclus::Context *__pyx_v_ctx) : cyclus::Agent(__pyx_v_ctx) {
                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/mouginot/work/app/cyclus/cyclus/build/cyclus/py_agents_.cxx:6166:99: error: only constructors take base initializers
void CyclusAgentShim::__pyx_f___init__CyclusAgentShim(CYTHON_UNUSED cyclus::Context *__pyx_v_ctx) : cyclus::Agent(__pyx_v_ctx) {
                                                                                                  ^
/Users/mouginot/work/app/cyclus/cyclus/build/cyclus/py_agents_.cxx:8410:24: error: out-of-line definition of '__pyx_f___init__CyclusRegionShim' does not match any
      declaration in 'CyclusRegionShim'
void CyclusRegionShim::__pyx_f___init__CyclusRegionShim(CYTHON_UNUSED cyclus::Context *__pyx_v_ctx) : cyclus::Region(__pyx_v_ctx) {
                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/mouginot/work/app/cyclus/cyclus/build/cyclus/py_agents_.cxx:8410:101: error: only constructors take base initializers
void CyclusRegionShim::__pyx_f___init__CyclusRegionShim(CYTHON_UNUSED cyclus::Context *__pyx_v_ctx) : cyclus::Region(__pyx_v_ctx) {
                                                                                                    ^
/Users/mouginot/work/app/cyclus/cyclus/build/cyclus/py_agents_.cxx:10780:29: error: out-of-line definition of '__pyx_f___init__CyclusInstitutionShim' does not match
      any declaration in 'CyclusInstitutionShim'
void CyclusInstitutionShim::__pyx_f___init__CyclusInstitutionShim(CYTHON_UNUSED cyclus::Context *__pyx_v_ctx) : cyclus::Institution(__pyx_v_ctx) {
                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/mouginot/work/app/cyclus/cyclus/build/cyclus/py_agents_.cxx:10780:111: error: only constructors take base initializers
void CyclusInstitutionShim::__pyx_f___init__CyclusInstitutionShim(CYTHON_UNUSED cyclus::Context *__pyx_v_ctx) : cyclus::Institution(__pyx_v_ctx) {
                                                                                                              ^
/Users/mouginot/work/app/cyclus/cyclus/build/cyclus/py_agents_.cxx:13159:26: error: out-of-line definition of '__pyx_f___init__CyclusFacilityShim' does not match any
      declaration in 'CyclusFacilityShim'
void CyclusFacilityShim::__pyx_f___init__CyclusFacilityShim(CYTHON_UNUSED cyclus::Context *__pyx_v_ctx) : cyclus::Facility(__pyx_v_ctx) {
                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/mouginot/work/app/cyclus/cyclus/build/cyclus/py_agents_.cxx:13159:105: error: only constructors take base initializers
void CyclusFacilityShim::__pyx_f___init__CyclusFacilityShim(CYTHON_UNUSED cyclus::Context *__pyx_v_ctx) : cyclus::Facility(__pyx_v_ctx) {
                                                                                                        ^
/Users/mouginot/work/app/cyclus/cyclus/build/cyclus/py_agents_.cxx:30444:49: warning: implicit conversion from 'long' to 'int' changes value from -9999999999 to
      -1410065407 [-Wconstant-conversion]
  __pyx_v_6cyclus_6agents__GET_MAT_PREFS_TIME = -9999999999;
                                              ~ ^~~~~~~~~~~
/Users/mouginot/work/app/cyclus/cyclus/build/cyclus/py_agents_.cxx:30476:50: warning: implicit conversion from 'long' to 'int' changes value from -9999999999 to
      -1410065407 [-Wconstant-conversion]
  __pyx_v_6cyclus_6agents__GET_PROD_PREFS_TIME = -9999999999;
                                               ~ ^~~~~~~~~~~
/Users/mouginot/work/app/cyclus/cyclus/build/cyclus/py_agents_.cxx:30508:48: warning: implicit conversion from 'long' to 'int' changes value from -9999999999 to
      -1410065407 [-Wconstant-conversion]
  __pyx_v_6cyclus_6agents__GET_MAT_BIDS_TIME = -9999999999;
                                             ~ ^~~~~~~~~~~
/Users/mouginot/work/app/cyclus/cyclus/build/cyclus/py_agents_.cxx:30540:49: warning: implicit conversion from 'long' to 'int' changes value from -9999999999 to
      -1410065407 [-Wconstant-conversion]
  __pyx_v_6cyclus_6agents__GET_PROD_BIDS_TIME = -9999999999;
                                              ~ ^~~~~~~~~~~
4 warnings and 8 errors generated.
make[2]: *** [cyclus/CMakeFiles/py_agents_.dir/py_agents_.cxx.o] Error 1
make[1]: *** [cyclus/CMakeFiles/py_agents_.dir/all] Error 2
make: *** [all] Error 2
```
this might be a known issue:
https://github.com/cython/cython/issues/1880
on only occurring on MACOS

**Comment by scopatz** [comment]
Hi @bam241, do you mean it fails against Cython 0.27 or Python 2.7?

**Comment by bam241** [comment]
Cython 0.27
(autocorrection sorry)

**Comment by scopatz** [comment]
What Python are you using? That thread seems to indicate that it might not happen on Python v3.5.2

**Comment by bam241** [comment]
python 3.5.4

**Comment by bam241** [comment]
my reference might be wrong, it seems to be an other issue.

**Comment by scopatz** [comment]
hmm interesting. I am still on Cython v0.26.1

**Comment by bam241** [comment]
I switch back to 0.26.1 and it is working.
They release 0.27 this morning or yesterday.

I had an issue with macport so I did a clean install and pip installed the last version by default...

(That’s why other issue about the dependencies— was a good way to check documentation)

BaM

**Comment by scopatz** [comment]
I see, well, I am glad that switching back worked for the moment. We'll have to fix this I imagine. I wonder if it breaks on linux too

**Comment by scopatz** [comment]
I just tried 0.27 on linux and it seemed to work fine

**Comment by scopatz** [comment]
Actually, nevermind, I am now seeing this issue on linux.  I didn't do a clean build.

**Comment by bam241** [comment]
@scopatz same problem on CI, the deployment part rely on `conda` for the dependencies, which updated the `Cython` version to 0.27.0:
https://circleci.com/gh/cyclus/cyclus/1314

**Comment by scopatz** [comment]
OK, well, I think I have a fix for the compile error on my `cy27` branch. However, there are a lot of segfaults in Cython functions now...

**Comment by bam241** [comment]
maybe we can stay on 0.26.1 for the moment ?

**Comment by scopatz** [comment]
See #1423

**Comment by bam241** [comment]
thx @scopatz closing,
fixed by #1423
